### PR TITLE
[headers] Move padmgr, scheduler, irqmgr symbols to respective headers; add segmented_address.h

### PR DESCRIPTION
**Pull request by Dragorn421**
Companion PR to https://github.com/zeldaret/oot/pull/2151

### CODE REVIEW

**Comment by Dragorn421** [comment]
this builds not OK, probably bss memes
I'm not familiar enough with MM to figure it out, any helper?

File: src/boot/idle.c [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#include "stdbool.h"
```
Comment (hensldm) [inline src/boot/idle.c:1]: Can you move this below the `prevent_bss_reordering`? We try to keep `prevent` as the first so it is easier to find and remove as needed.

**Review by hensldm — COMMENTED** [review]
Also as mentioned in discord, could you move the `gIrqMgr` extern to `irqmgr.h`? If you have more bss issues feel free to reach out.

**Review by AngheloAlf — APPROVED** [review]
Hopefully bss won't be that terrible to fix

**Comment by Dragorn421** [comment]
also added segmented_address.h , keeping up with the oot counterpart PR

File: include/segmented_address.h
Comment (AngheloAlf) [inline include/segmented_address.h:1]: I wonder if we should merge this header with `segment_symbols.h`, but arguably we should get rid of the latter so it is fine this way

**Review by AngheloAlf — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

File: include/segmented_address.h
Comment (cadmic) [reply include/segmented_address.h:1]: segmented addresses and ROM segments are fairly separate concepts though. It's true that some ROM segments are placed at a segmented address but do not be deceived

**Pull request merged**